### PR TITLE
refactor(utils.py): move language data to JSON file (resolves #52)

### LINKED ISSUE #52: Refactor language utility data out from utils.py
### Terms

- [X] I have searched [open and closed issues](https://github.com/scribe-org/Scribe-Data/issues?q=is%3Aissue)
- [X] I agree to follow Scribe-Data's [Code of Conduct](https://github.com/scribe-org/Scribe-Data/blob/main/.github/CODE_OF_CONDUCT.md)

### Issue

This issue is for refactoring the language data out from [src/scribe_data/utils.py](https://github.com/scribe-org/Scribe-Data/blob/main/src/scribe_data/utils.py) to a separate file, likely a JSON - as suggested by @m-charlton in #51 :raised_hands: 

As proposed by @m-charlton, 
> This file would be loaded once on module import and then the utils functions could interrogate this loaded object.

We can use this issue to track discussion of details for the implementation and the work itself.

### PR DESCRIPTION
**Pull request by m-charlton**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description
Moves language meta data from python source code into an external JSON resource which is loaded when the `utils` module is imported.

#### `language-meta-data.json`
Please place attention to the comment properties: "description", "iso", "qid" & "remove-words".

#### `utils.py`
Couldn't think of a better name for the `_find` function - suggestions welcome.

 

- #52

### CODE REVIEW

File: src/scribe_data/utils.py [hunk 1]
```diff
@@ -230,7 +241,7 @@ def get_path_from_format_file() -> str:
     return "../../../../../.."
 
 
-def get_path_from_load_dir() -> str:
+def get_path_from_load_dir():
```
Comment (m-charlton) [inline src/scribe_data/utils.py:244]: Missing return type annotation: `str`

File: src/scribe_data/utils.py [hunk 2]
```diff
@@ -120,30 +175,21 @@ def get_language_from_iso(iso: str) -> str:
...
+        "language",
+        f"{iso.upper()} is currently not a supported ISO language.",
+    ).capitalize()
 
 
 def get_language_words_to_remove(language: str) -> list[str]:
     """
-    Returns the words that should not be included as autosuggestions for the given language.
+    Returns the words that should not be included as autosuggestions for the given
+    language.
```
Comment (m-charlton) [inline src/scribe_data/utils.py:192]: No new line (combine with previous line)

File: src/scribe_data/utils.py [hunk 3]
```diff
@@ -152,46 +198,22 @@ def get_language_words_to_remove(language: str) -> list[str]:
 
     Returns
     -------
-        The words that should not be included as autosuggestions for the given language as values of a dictionary.
+        list[str]
+            The words that should not be included as autosuggestions for the given
+            language
```
Comment (m-charlton) [inline src/scribe_data/utils.py:203]: No new line (combine with previous line)

File: src/scribe_data/utils.py [hunk 4]
```diff
@@ -200,27 +222,16 @@ def get_language_words_to_ignore(language: str) -> list[str]:
...
```
Comment (m-charlton) [inline src/scribe_data/utils.py:227]: No new line (combine with previous line)

File: src/scribe_data/utils.py [hunk 5]
```diff
@@ -349,7 +363,8 @@ def check_and_return_command_line_args(
     all_args, first_args_check=None, second_args_check=None
 ):
     """
-    Checks command line arguments passed to Scribe-Data files and returns them if correct.
+    Checks command line arguments passed to Scribe-Data files and returns them if
```
Comment (m-charlton) [inline src/scribe_data/utils.py:366]: No new line (combine with previous line)

File: src/scribe_data/utils.py [hunk 6]
```diff
@@ -365,7 +380,8 @@ def check_and_return_command_line_args(
     Returns
     -------
         first_args, second_args: list(str)
-            The subset of possible first and second arguments that have been verified as being valid.
+            The subset of possible first and second arguments that have been verified
+            as being valid.
```
Comment (m-charlton) [inline src/scribe_data/utils.py:384]: No new line (combine with previous line)

File: src/scribe_data/utils.py [hunk 7]
```diff
@@ -152,46 +198,22 @@ def get_language_words_to_remove(language: str) -> list[str]:
...
+    return _find(
+        "language",
+        language,
+        "remove-words",
+        f"{language.capitalize()} is currently not a supported language.",
+    )
 
 
 def get_language_words_to_ignore(language: str) -> list[str]:
     """
```
Comment (m-charlton) [inline src/scribe_data/utils.py:214]: @andrewtavis : This function has the same description as `get_language_words_to_remove` on line 189. This _could_ be an error. One this is rectified the entry on line 9 of `src/scribe_data/resources/language_meta_data.json` can be updated.

File: src/scribe_data/utils.py [hunk 8]
```diff
@@ -22,23 +22,102 @@
...
-        "English",
-        "French",
-        "German",
-        "Italian",
-        "Portuguese",
-        "Russian",
-        "Spanish",
-        "Swedish",
-    ]
+    return sorted(entry["language"].capitalize() for entry in _languages)
```
Comment (andrewtavis) [inline src/scribe_data/utils.py:120]: Very nice :)

File: src/scribe_data/utils.py [hunk 9]
```diff
@@ -52,27 +131,15 @@ def get_language_qid(language: str) -> str:
...
-        "swedish": "Q9027",
-    }
-
-    if language not in language_qid_dict:
-        raise ValueError(
-            f"{language.upper()} is currently not a supported language for QID conversion."
-        )
-
-    return language_qid_dict[language]
+    return _find(
```
Comment (andrewtavis) [inline src/scribe_data/utils.py:137]: This is really cool 🤩

File: src/scribe_data/utils.py [hunk 10]
```diff
@@ -86,27 +153,15 @@ def get_language_iso(language: str) -> str:
...
-        "swedish": "sv",
-    }
-
-    if language not in language_iso_dict:
-        raise ValueError(
-            f"{language.capitalize()} is currently not a supported language for ISO conversion."
-        )
-
-    return language_iso_dict[language]
+    return _find(
```
Comment (andrewtavis) [inline src/scribe_data/utils.py:159]: This is what I'm talking about in #55. We could basically get rid of this function and use langcodes, thus allowing us to add languages without needing to track their ISOs :)

File: src/scribe_data/utils.py [hunk 11]
```diff
@@ -120,30 +175,21 @@ def get_language_from_iso(iso: str) -> str:
...
-        "ru": "Russian",
-        "es": "Spanish",
-        "sv": "Swedish",
-    }
-
-    if iso not in iso_language_dict:
-        raise ValueError(f"{iso.upper()} is currently not a supported ISO language.")
-
-    return iso_language_dict[iso]
+    return _find(
```
Comment (andrewtavis) [inline src/scribe_data/utils.py:181]: Similarly for this, we could remove it with #55 :)

File: src/scribe_data/utils.py [hunk 12]
```diff
@@ -255,7 +266,8 @@ def get_ios_data_path(language: str) -> str:
 
     Returns
     -------
-        The path to the data json for the given language.
+        str
```
Comment (andrewtavis) [inline src/scribe_data/utils.py:269]: Thanks for these 🙏 Definitely is good standard :)

**Review by andrewtavis — APPROVED** [review]
Really great, @m-charlton! Thanks for all this :) I'll do a quick sweep to get any minor fixes that remain 😊

File: src/scribe_data/utils.py (hunk 4 above)
Comment (andrewtavis) [reply src/scribe_data/utils.py:227]: Thank you! Fixed locally and will send it shortly :)

**Pull request merged**